File: test/Transcript_test.java
import org.junit.jupiter.api.Test;

import java.util.ArrayList;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class Transcript_test {
    @Test
    public void calculateGPA_test() {
        DataManagement dataManagement = new DataManagement();
        Student student = dataManagement.getStudent("baristokgoz@marun.edu.tr");
        Transcript transcript = student.getTranscript();
        float trueGPA = 1.8125f;
        float calculatedGPA = transcript.get_GPA();
        assertEquals(trueGPA, calculatedGPA, "GPA should be the same!");


    }

}

File: Test/Transcript_test.py
import unittest

from pythonCode.DataManagement import DataManagement


class Transcript_test(unittest.TestCase):
    def test_calculateGPA_test(self):
        data_management = DataManagement()
        student = data_management.getStudent("selinpolat@marun.edu.tr")
        transcript = student.getTranscript()
        trueGPA = 2.5
        calculatedGPA = transcript.get_GPA()
        self.assertEqual(trueGPA, calculatedGPA, "GPA should be the same!")